test-data/unit/check-functions.test

```diff
@@ -2125,3 +2125,82 @@ def i() -> List[Union[str, int]]:
     return x
 
 [builtins fixtures/dict.pyi]
+
+[case testAnyStrIncompatibleArguments]
+from typing import TypeVar
+AnyStr = TypeVar('AnyStr', str, bytes)
+def f(x: AnyStr, y: AnyStr) -> None: pass
+def g(x: AnyStr, y: AnyStr, z: int) -> AnyStr: pass
+f('a', 'b')
+f(b'a', b'b')
+f('a', b'b') # E: Type argument 1 of "f" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+g('a', 'b', 1)
+g(b'a', b'b', 1)
+g('a', b'b', 1) # E: Type argument 1 of "g" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+g('a', b'b', 'c') # E: Type argument 1 of "g" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes \
+# E: Argument 3 to "g" has incompatible type "str"; expected "int"
+
+[case testUnionAnyStrIncompatibleArguments]
+from typing import TypeVar, Union
+AnyStr = TypeVar('AnyStr', str, bytes)
+def f(x: Union[AnyStr, int], y: AnyStr) -> None: pass
+f('a', 'b')
+f(1, 'b')
+f('a', b'b') # E: Type argument 1 of "f" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+
+[case testStarAnyStrIncompatibleArguments]
+from typing import TypeVar, Union
+AnyStr = TypeVar('AnyStr', str, bytes)
+def f(*x: AnyStr) -> None: pass
+def g(x: int, *y: AnyStr) -> None: pass
+def h(*x: AnyStr, y: int) -> None: pass
+f('a')
+f('a', 'b')
+f('a', b'b') # E: Type argument 1 of "f" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+g(1, 'a')
+g(1, 'a', b'b') # E: Type argument 1 of "g" has incompatible value \
+# N: "AnyStr" must be all one type: str or bytes
+h('a', y=1)
+h('a', 'b', y=1)
+h('a', b'b', y=1) # E: Type argument 1 of "h" has incompatible value "object"
```

Comment:
They are described in ``nodes.py`` above ``class CallExpr...``. I copy here for convenience:
```python
# Positional argument
ARG_POS = 0  # type: int
# Positional, optional argument (functions only, not calls)
ARG_OPT = 1  # type: int
# *arg argument
ARG_STAR = 2  # type: int
# Keyword argument x=y in call, or keyword-only function arg
ARG_NAMED = 3  # type: int
# **arg argument
ARG_STAR2 = 4  # type: int
# In an argument list, keyword-only and also optional
ARG_NAMED_OPT = 5
```
If this will take time, then maybe we need to postpone this for a separate PR. I don't want this one to take long.